File: app/src/main/java/com/volkanhotur/basemvvm/android/base/activity/BaseBindingActivity.java
package com.volkanhotur.basemvvm.android.base.activity;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.databinding.ViewDataBinding;

import com.core.abstractbase.AbstractBindingActivity;
import com.volkanhotur.basemvvm.android.base.exception.UnCaughtExceptionHandler;
import com.volkanhotur.basemvvm.android.utils.impl.DefaultClickHandler;
import com.volkanhotur.basemvvm.android.utils.helper.LocaleHelper;

/**
 * @author volkanhotur
 */

public abstract class BaseBindingActivity< VDB extends ViewDataBinding> extends AbstractBindingActivity< VDB> implements DefaultClickHandler {

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(LocaleHelper.onAttach(base));
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        Thread.setDefaultUncaughtExceptionHandler(new UnCaughtExceptionHandler(this));
        super.onCreate(savedInstanceState);
    }

    @Override
    public void navigateView() { }

    @Override
    public void expireSession() {}

    @Override
    public boolean isLoadingBarEnabled() {
        return false;
    }
}

File: app/src/main/java/com/volkanhotur/basemvvm/android/base/activity/BaseBindingActivity.kt
package com.volkanhotur.basemvvm.android.base.activity

import android.content.Context
import android.os.Bundle
import androidx.databinding.ViewDataBinding
import com.core.abstractbase.AbstractBindingActivity
import com.volkanhotur.basemvvm.android.base.exception.UnCaughtExceptionHandler
import com.volkanhotur.basemvvm.android.utils.helper.LocaleHelper
import com.volkanhotur.basemvvm.android.utils.impl.DefaultClickHandler

/**
 * @author volkanhotur
 */
abstract class BaseBindingActivity<VDB : ViewDataBinding?> : AbstractBindingActivity<VDB>(), DefaultClickHandler {

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(LocaleHelper.onAttach(base))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        Thread.setDefaultUncaughtExceptionHandler(UnCaughtExceptionHandler(this))
        super.onCreate(savedInstanceState)
    }

    override fun navigateView() {}

    override fun expireSession() {}

    override val isLoadingBarEnabled: Boolean
        get() = false
}